[Safe CPP] Address Warnings in Quad.h

stwrt opened the pull request:
#### f56307b11b40ab389f4acf8cb24a532284a430e2
<pre>
[Safe CPP] Address Warnings in Quad.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=294000">https://bugs.webkit.org/show_bug.cgi?id=294000</a>
<a href="https://rdar.apple.com/problem/152549205">rdar://problem/152549205</a>

Reviewed by Charlie Wolfe.

Address safe cpp warnings in Quad.h. This also fixes css/DeprecatedCSSOMRect.h and css/Rect.h.

 * Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
 * Source/WebCore/css/CSSPropertyInitialValues.cpp:
 (WebCore::isNumber):
 * Source/WebCore/css/RectBase.h:
 (WebCore::RectBase::left const):
 (WebCore::RectBase::protectedTop const): Deleted.
 (WebCore::RectBase::protectedRight const): Deleted.
 (WebCore::RectBase::protectedBottom const): Deleted.
 (WebCore::RectBase::protectedLeft const): Deleted.

Canonical link: <a href="https://commits.webkit.org/295818@main">https://commits.webkit.org/295818@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7c14b7c26cfa2c8301a4af9cb0f8403aeac09b5a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106232 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25981 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16376 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111430 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56828 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108271 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26643 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34484 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80694 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109236 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21070 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95863 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61021 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20607 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13964 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56268 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90426 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13999 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114290 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33370 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24605 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89766 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33734 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92094 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89467 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22815 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34337 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12149 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28950 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33295 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38707 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33041 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36391 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34639 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe reviewed (changes requested):
Instead of doing this, you should remove each `protectedX()` function on RectBase and make m_top, m_bottom, m_right and m_left const.

https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#when-calling-a-non-trivial-member-function-on-an-object-hold-a-smart-pointer-to-the-object-on-the-stack

stwrt committed:
[Safe CPP] Address Warnings in Quad.h
https://bugs.webkit.org/show_bug.cgi?id=294000
rdar://problem/152549205

Reviewed by Charlie Wolfe.

Address safe cpp warnings in Quad.h. This also fixes css/DeprecatedCSSOMRect.h and css/Rect.h.

 * Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
 * Source/WebCore/css/CSSPropertyInitialValues.cpp:
 (WebCore::isNumber):
 * Source/WebCore/css/RectBase.h:
 (WebCore::RectBase::left const):
 (WebCore::RectBase::protectedTop const): Deleted.
 (WebCore::RectBase::protectedRight const): Deleted.
 (WebCore::RectBase::protectedBottom const): Deleted.
 (WebCore::RectBase::protectedLeft const): Deleted.

Canonical link: https://commits.webkit.org/295818@main
